lib/compress/zstd_compress.c

```diff
@@ -4060,19 +4060,21 @@ size_t ZSTD_compress2(ZSTD_CCtx* cctx,
                       void* dst, size_t dstCapacity,
                       const void* src, size_t srcSize)
 {
-    size_t oPos = 0;
-    size_t iPos = 0;
-    size_t const result = ZSTD_compressStream2_simpleArgs(cctx,
-                                    dst, dstCapacity, &oPos,
-                                    src, srcSize, &iPos,
-                                    ZSTD_e_end);
-    assert(iPos == srcSize);
-    if (ZSTD_isError(result)) return result;
-    if (result != 0) {  /* compression not completed, due to lack of output space */
-        assert(oPos == dstCapacity);
-        return ERROR(dstSize_tooSmall);
+    ZSTD_CCtx_reset(cctx, ZSTD_reset_session_only);
+    {   size_t oPos = 0;
+        size_t iPos = 0;
+        size_t const result = ZSTD_compressStream2_simpleArgs(cctx,
+                                        dst, dstCapacity, &oPos,
+                                        src, srcSize, &iPos,
+                                        ZSTD_e_end);
+        if (ZSTD_isError(result)) return result;
+        assert(iPos == srcSize);
```

Comment:
Oops, does this also need to go after `result != 0`?

Reply:
Indeed !
if result is non-zero, we expect failure due to insufficient `dstCapacity`,
so there is no guarantee that input was entirely consumed !